engine/src/test/java/org/hibernate/validator/test/internal/constraintvalidators/hv/SafeHtmlValidatorTest.java

```diff
@@ -164,6 +164,20 @@ public void testAdditionalProtocols() {
 		assertNumberOfViolations( validator.validate( new Bar( "<div src='data:image/png;base64,100101' />" ) ), 1 );
 	}
 
+	@Test
+	@TestForIssue(jiraKey = "HV-817")
+	public void testPreserveRelativeLinks() throws Exception {
+		descriptor.setValue( "whitelistType", WhiteListType.RELAXED );
+		descriptor.setValue( "baseUri", "http://127.0.0.1" );
+
+		assertTrue( getSafeHtmlValidator().isValid( "<img src='/some/relative/url/image.png' />", null ) );
```

Comment:
Yeah, let's be safe.

@marko-bekhta can you address the minor comments I made? Then we will merge it.

As for backporting it, as mentioned by Gunnar, we would only do it if we do a new 5.x release. For now we don't really have a lot of material to justify a new 5.x release and the associated maintenance burden but we'll see in the future. Using the next alpha of HV 6 would be an option if you use JDK 8.